### DISCUSSION TITLE
Two Interdependent select2 fields reset each other?

### DISCUSSION THREAD

**Discussion by MiroJelev** [codingjoe/django-select2#230] [Q&A]
**Goal**
I want to make filters for a table that inter depend on each other so that it doesn't matter which filter I choose to start with first.

**Problem**
The problem comes when I set the filters to depend on each other in the `dependent_fields` like so:
https://django-select2.readthedocs.io/en/latest/extra.html#interdependent-select2
When I select the first field, everything is okay, but when I click a new value for the second field, it resets (clears) the first field.  The same happens when I try the other way around.
Am I missing something?

Cheers!

**Code Snippet**

```python
class Test1Widget(s2forms.ModelSelect2Widget):
    search_fields = [
        "code__icontains"
    ]
    dependent_fields = {
        "test2": "db__test2",
    }

class Test2Widget(s2forms.ModelSelect2Widget):
    search_fields = [
        "id__icontains"
    ]
    dependent_fields={
        "test1": "db__test1",
    }
test1 = ModelChoiceFilter(queryset=test1.objects.all(), widget=Test1Widget)
test2 = ModelChoiceFilter(queryset=test2.objects.all(), widget=Test2Widget)

```

**Comment by codingjoe** [comment]
Hi @MiroJelev,

Thank you for reaching out. Honestly, I haven't considered this use case yet. Interdependent fields are not sound like they are going to introduce a lot of trouble, on a database level or using select2.

Would you mind sharing the exact use case? Every so often, the best solution isn't to add a patch, but to rethink the implementation. In a "the best part is no part" kinda way.

If you want to drop the clearing behavior, please you can provide your custom JS copy, while dropping the `val('')` in this line https://github.com/codingjoe/django-select2/blob/5753384b489c283b7b08f9f4346c80a5a735fbe6/django_select2/static/django_select2/django_select2.js#L63

Cheers!
Joe
